### PR TITLE
Fixed 6287 - Added support for pause/no-pause in pipeline config API …

### PR DESCRIPTION
**Pull request by msaurabh1978**
Fixed Issue: gocd/gocd#6287 (Added ability to specify whether the pipeline should be paused on create.

Changes Done are,

- [x] Updated API doc for pipeline config create API to include request headers that allow specifying 1) whether the pipeline should be paused on creating and 2) cause for pausing the pipeline on creating.

- [x] Updated sample API request for creating a pipeline to include request header `X-pause-pipeline-after-create` that allows specifying pause behavior of the pipeline and to include request header `X-pause-cause` that allows specifying the cause for pausing the pipeline on creating.

### CODE REVIEW

**Comment by maheshp** [comment]
We need to bump up the API version to support the new header. Hold this PR till the v8 of the Pipeline Config API is implemented.

File: source/includes/pipeline_config/_031-create.md.erb [hunk 1]
```diff
@@ -241,6 +243,16 @@ Content-Type: <%= data.apis.versions.pipeline %>; charset=utf-8
 
 Create a pipeline.
 
+<aside class="">
```
Comment (ketan) [inline source/includes/pipeline_config/_031-create.md.erb]: Provide a class?

File: source/includes/pipeline_config/_031-create.md.erb [hunk 2]
```diff
@@ -241,6 +243,16 @@ Content-Type: <%= data.apis.versions.pipeline %>; charset=utf-8
...
+  <strong>Note:</strong>
+  The create pipeline config API supports an optional header parameter <code>X-pause-pipeline-after-create</code> that
+  allow you to specify whether the pipeline should be paused after create or not. If this header parameter is missing or
+  contains a value other than <i>true</i>, then the created pipeline will not be paused.
```
Comment (ketan) [inline source/includes/pipeline_config/_031-create.md.erb]: Use `<code/>` not `<i/>`. See https://softwareengineering.stackexchange.com/questions/255366/why-are-the-b-and-i-tags-deprecated.

Also `header parameter` should be called `header field` instead. I'd probably rephrase this as:

> The create pipeline config API supports an optional header field `X-pause-pipeline`. This field, if set to `true`, will create the pipeline in the `paused` state. The additional header field `X-pause-cause` may optionally contain the reason for pausing the pipeline.
